### DISCUSSION TITLE
Metaobject Page references return IDs in incorrect format

### DISCUSSION THREAD

**Discussion by duncan-fairley** [Shopify/storefront-api-feedback#178] [Help] [answered]
In the following query, I am unable to traverse a Page metafield reference (behavior is consistent on `2023-01`, `2023-04` and `unstable`)

```
query {
  metaobject (id:"gid://shopify/Metaobject/148635783"){
      homepage:field(key:"homepage") {
        value
        reference {
          ...on Page {
            title 
          }
        }
      }
  }
}
```
Returns:
```
{
   "data": {
      "metaobject": {
         "homepage": {
            "value": "gid://shopify/OnlineStorePage/84283064455",
            "reference": null
         }
      }
   }
}
```

It seems that the ID returned is in the incorrect format. When I run the following query with the value returned by the previous query:

```
query {
   page(id: "gid://shopify/OnlineStorePage/84283064455") {
      title
   }
}
```
I get the following error:
```
{
   "data": {
      "page": null
   },
   "errors": [
      {
         "message": "invalid id",
         "locations": [
            {
               "line": 2,
               "column": 4
            }
         ],
         "path": [
            "page"
         ]
      }
   ]
}
```

However, the following query works just fine:
```
query {
   page(id: "gid://shopify/Page/84283064455") {
      title
   }
}
```

**Comment by johncraigcole** [comment, marked as answer]
Hey @DuncanFa we shipped a fix for this yesterday. Let me know if you're still experiencing a problem with a page reference.

**Reply by duncan-fairley** [reply]
Working great @johncraigcole, thanks for the quick resolution on this!

Page ID is still returning in the `/OnlineStorePage/` type but this isn't a problem for us at the moment. 

Appreciate the help

**Reply by johncraigcole** [reply]
Strange, I'm seeing `gid://shopify/Page/` on a new meta object page field reference. If you could share an `x-request-id` for the SFAPI request where you're still seeing `gid://shopify/OnlineStorePage` returned that would be helpful. I'd also be curious if you created a new meta object entry with a page field reference if that also is returning `gid://shopify/OnlineStorePage/`

**Reply by duncan-fairley** [reply]
@johncraigcole - here's a request ID: `364af2a6-1142-4501-9080-5f2fe33c6d17`

Testing with a newly created reference now

**Reply by duncan-fairley** [reply]
@johncraigcole - Getting `gid://shopify/OnlineStorePage/` on newly created reference as well: `2056c9c0-801b-4191-b95e-ba88320eba2b`

**Reply by johncraigcole** [reply]
Hey @DuncanFa thanks for the examples. So the null on the reference was got fixed, and yes the `value` attribute of the field is still showing with `OnlineStorePage`. The fix for the `reference` attribute ensures a compatible Storefront API id for a page record, and would recommend using the `reference` field for the id and any other specifics about the referenced type. 

```
field(key:"special_page"){
	key
	value
	reference{
		... on Page{
			id					
			title
		}
	}
}
```

**Reply by duncan-fairley** [reply]
> Hey @DuncanFa thanks for the examples. So the null on the reference was got fixed, and yes the `value` attribute of the field is still showing with `OnlineStorePage`. The fix for the `reference` attribute ensures a compatible Storefront API id for a page record, and would recommend using the `reference` field for the id and any other specifics about the referenced type. 
> 
> 
> 
> ```
> 
> field(key:"special_page"){
> 
> 	key
> 
> 	value
> 
> 	reference{
> 
> 		... on Page{
> 
> 			id					
> 
> 			title
> 
> 		}
> 
> 	}
> 
> }
> 
> ```
> 
> 
> 
> 
> 
> 

That makes sense and is not an issue at the moment for us. Thanks again!
